### PR TITLE
WebRTCSession: avoid deprecated 'gst_element_get_request_pad'

### PR DESCRIPTION
**Pull request by sc0w**

### CODE REVIEW

File: src/voip/WebRTCSession.cpp [hunk 1]
```diff
@@ -32,6 +32,10 @@ extern "C"
 }
 #endif
 
+#if !GST_CHECK_VERSION(1, 20, 0)
+#define gst_element_request_pad_simple gst_element_get_request_pad
+#endif
+
```
Comment (deepbluev7) [inline src/voip/WebRTCSession.cpp:38]:
```suggestion
}

#if !GST_CHECK_VERSION(1, 20, 0)
#define gst_element_request_pad_simple gst_element_get_request_pad
#endif

#endif
```

Comment (sc0w) [reply src/voip/WebRTCSession.cpp:38]: ok, or do you prefer these changes instead?

```patch
diff --git a/src/voip/WebRTCSession.cpp b/src/voip/WebRTCSession.cpp
index 50fe2539..68fd8799 100644
--- a/src/voip/WebRTCSession.cpp
+++ b/src/voip/WebRTCSession.cpp
@@ -387,7 +387,11 @@ addLocalPiP(GstElement *pipe, const std::pair<int, int> &videoCallSize)
     gst_object_unref(tee);
 
     GstElement *compositor = gst_bin_get_by_name(GST_BIN(pipe), "compositor");
+#if GST_CHECK_VERSION(1, 20, 0)
+    localPiPSinkPad_       = gst_element_request_pad_simple(compositor, "sink_%u");
+#else
     localPiPSinkPad_       = gst_element_get_request_pad(compositor, "sink_%u");
+#endif
     g_object_set(localPiPSinkPad_, "zorder", 2, nullptr);
 
     bool isVideo         = WebRTCSession::instance().callType() == CallType::VIDEO;
@@ -435,7 +439,11 @@ addLocalVideo(GstElement *pipe)
 {
     GstElement *queue = newVideoSinkChain(pipe);
     GstElement *tee   = gst_bin_get_by_name(GST_BIN(pipe), "videosrctee");
+#if GST_CHECK_VERSION(1, 20, 0)
+    GstPad *srcpad    = gst_element_request_pad_simple(tee, "src_%u");
+#else
     GstPad *srcpad    = gst_element_get_request_pad(tee, "src_%u");
+#endif
     GstPad *sinkpad   = gst_element_get_static_pad(queue, "sink");
     if (GST_PAD_LINK_FAILED(gst_pad_link(srcpad, sinkpad)))
         nhlog::ui()->error("WebRTC: failed to link videosrctee -> video sink chain");
@@ -955,7 +963,11 @@ WebRTCSession::addVideoPipeline(int vp8PayloadType)
             }
 
             GstPad *srcpad    = gst_element_get_static_pad(camerafilter, "src");
+#if GST_CHECK_VERSION(1, 20, 0)
+            remotePiPSinkPad_ = gst_element_request_pad_simple(compositor, "sink_%u");
+#else
             remotePiPSinkPad_ = gst_element_get_request_pad(compositor, "sink_%u");
+#endif
             if (GST_PAD_LINK_FAILED(gst_pad_link(srcpad, remotePiPSinkPad_))) {
                 nhlog::ui()->error("WebRTC: failed to link camerafilter -> compositor");
                 gst_object_unref(srcpad);
```

Comment (deepbluev7) [reply src/voip/WebRTCSession.cpp]: No, I think what you did is pretty reasonable, it just didn't compile, when there is no gstreamer :3

Comment (sc0w) [reply src/voip/WebRTCSession.cpp]: ok, done with your suggested changes

**Comment by deepbluev7** [comment]
Okay, thank you, looks good to me!

**Pull request merged**